torch/_dynamo/variables/builtin.py

```diff
@@ -1665,10 +1667,34 @@ def check_type(ty):
                 isinstance_type.__class__.__instancecheck__(isinstance_type, arg.value)
             )
 
+        isinstance_type_tuple: tuple[type, ...]
+        if isinstance(isinstance_type, type) or callable(
+            # E.g. isinstance([], typing.Sequence)
+            getattr(isinstance_type, "__instancecheck__", None)
```

Comment:
Couldn't this call arbirary code at compile time?  Do we need to trace through it?

Reply:
> Couldn't this call arbirary code at compile time?

Yes, it can. But I'm not sure we need to support this. The previous implementation before this PR already can run an arbitrary `__subclasscheck__` method.

> Do we need to trace through it?

Trace through this can have intensive overheads of `isinstance(...)` which is really common in user code. Also, most `__subclasscheck__` can be constant fold through, which degenerates to `issubclass(vt1.as_python_constant(), vt2.as_python_constant())`.